packages/flutter/lib/src/material/input.dart

```diff
@@ -79,12 +188,29 @@ class _InputState extends State<Input> {
 
   GlobalKey get focusKey => config.key is GlobalKey ? config.key : _rawInputLineKey;
 
+  // Optional state to retain if we are inside a Form widget.
+  _FormFieldData _formData = null;
+
```

Comment:
Should I name this back to _FormFieldState? It doesn't derive from State, but it is used only as a member var of InputState. And it does represent the 'state' of the form field...